### PR TITLE
[FEATURE] Add registration form hooks

### PR DESCRIPTION
**Pull request by mk-mxp**
As requested by @oliverklee in #241

* Hooks to header, form and footer
* Test for the hooks

I will add the changelog entry when this is almost finished (to prevent conflicts when rebasing).

### CODE REVIEW

**Comment by mk-mxp** [comment]
I don't understand the failing of this test (actually there was no change to code involved): `This ghost was created via getNewGhost and must not be loaded`

**Comment by oliverklee** [comment]
> I don't understand the failing of this test (actually there was no change to code involved)

This probably is due to some existing records in your DB. I'll need to move the corresponding test to the nimut TF as well to make the test stable again. #384

**Comment by oliverklee** [comment]
Oh, I take that back. If the test fails on Travis, this cannot have anything to do with existing records in the DB.

**Comment by oliverklee** [comment]
I'll look into the failing test.

**Comment by oliverklee** [comment]
@mk-mxp I have moved that test to nimut/tf. Could you please rebase?

File: Tests/LegacyUnit/FrontEnd/DefaultControllerTest.php [hunk 1]
```diff
@@ -7475,6 +7476,43 @@ public function testListOfRequirementsForEventWithTwoNotFulfilledRequirementsSho
...
+                'object_type' => \Tx_Seminars_Model_Event::TYPE_COMPLETE,
+                'title' => 'Registration form test',
+                'begin_date' => $GLOBALS['SIM_EXEC_TIME'] + 1000,
+                'end_date' => $GLOBALS['SIM_EXEC_TIME'] + 2000,
+                'needs_registration' => 1,
+                'attendees_max' => 10,
+            ]
+        );
+
+        $this->subject->piVars['seminar'] = $eventUid;
```
Comment (oliverklee) [inline Tests/LegacyUnit/FrontEnd/DefaultControllerTest.php]: Cast to string here, please.

**Comment by mk-mxp** [comment]
Thanks for cleaning up the tests!

**Pull request merged**